src/R/Editor/Application.Test/Completion/IntellisenseTest.cs

```diff
@@ -373,35 +373,28 @@ public class IntellisenseTest : FunctionIndexBasedTest {
             }
         }
 
-        [Test]
+        [Test(ThreadType = ThreadType.UI)]
         public async Task R_SnippetsCompletion01() {
-            using (var script = await _editorHost.StartScript(ExportProvider, RContentTypeDefinition.ContentType)) {
+            using (var script = await _editorHost.StartScript(Shell, RContentTypeDefinition.ContentType)) {
                 script.DoIdle(100);
                 script.Type("whil");
                 script.DoIdle(300);
 
-                UIThreadHelper.Instance.Invoke(() => {
-                    var session = script.GetCompletionSession();
-                    session.Should().NotBeNull();
-
-                    var infoSourceProvider = ExportProvider.GetExportedValue<ISnippetInformationSourceProvider>();
-                    var infoSource = infoSourceProvider.InformationSource;
-                    var completion = session.SelectedCompletionSet.SelectionStatus.Completion;
+                var session = script.GetCompletionSession();
```

Comment:
Technically any test that uses some StartScript or editor host should be UI only.